### PR TITLE
Fix/filters vs settings separation [ROAD-752]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
### Fixed
- Unify names for Types of Scan (Snyk Products) in different places of UI
- Other minor UI improvements (including updated Severity icons) 

### Changed
- Separation of **Filtering**(for results in the Tree) and **Enablement**(in the Settings)for: 
  - Types of Scan (Snyk Products)
  - Severities.

image
image
image

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt [hunk 1]
```diff
@@ -39,39 +40,58 @@ class ScanTypesPanel(
...
                 checkBox(
-                    "Snyk Open Source vulnerabilities",
-                    { settings.ossScanEnable },
-                    { settings.ossScanEnable = it },
-                    cliScanComments
-                )
-                label("").component.convertIntoHelpHintLabel(
-                    "Find and automatically fix open source vulnerabilities"
-                )
+                    text = ProductType.OSS.productSelectionName,
```
[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt:75]: Actually IntelliJ platform provide quite a good api for that. They use it internally for i18n of Idea as well.
But imho that's not our top priority task for now... I actually don't think there are too many developers who doesn't know English, at least on level to get UI texts.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/TreePanel.kt [hunk 2]
```diff
@@ -18,11 +18,10 @@ class TreePanel(tree: Tree) : SimpleToolWindowPanel(true, true) {
     init {
         name = "treePanel"
         val severityToolbarPanel = JPanel(BorderLayout())
-        severityToolbarPanel.add(JLabel("Severity: "), BorderLayout.WEST)
+        severityToolbarPanel.add(JLabel("  Severity: "), BorderLayout.WEST)
```
[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/TreePanel.kt:21]: Yep, to make proper alignment :) 
probably doing left border would be more proper way but needed some investigation inside nested swing components here...

File: src/main/kotlin/snyk/container/ui/BaseImageRemediationDetailPanel.kt [hunk 3]
```diff
@@ -158,7 +161,7 @@ class BaseImageRemediationDetailPanel(
         )
 
         panel.isOpaque = true
-        panel.background = Severity.getBgColor(severity)// UIUtil.mix(Color.WHITE, baseColor, 0.25)
+        panel.background = severity.getBgColor()// UIUtil.mix(Color.WHITE, baseColor, 0.25)
```
[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/main/kotlin/snyk/container/ui/BaseImageRemediationDetailPanel.kt:164]: alternative funny/hacky way to do background color :)

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt [hunk 4]
```diff
@@ -955,15 +962,16 @@ class SnykToolWindowPanel(val project: Project) : JPanel(), Disposable {
...
 
-        if (pluginSettings().containerScanEnabled && containerResult.allCliIssues != null) {
+        val settings = pluginSettings()
+        if (settings.containerScanEnabled && settings.treeFiltering.containerResults && containerResult.allCliIssues != null) {
             containerResult.allCliIssues!!.forEach { issuesForImage ->
                 if (issuesForImage.vulnerabilities.isNotEmpty()) {
                     val imageTreeNode = ContainerImageTreeNode(issuesForImage, project)
                     rootContainerIssuesTreeNode.add(imageTreeNode)
 
                     issuesForImage.vulnerabilities
```
[the remark this answers is not in the document]
Comment (michelkaporin) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:972]: This file is more than 1000 LOC, which probably means it has too much responsibility or repeats many things. Just flagging it here, but feel free to leave it as-is, if that's the panel's best shape 👌

[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:972]: This our "god-class" :( and the ticket to refactor it, is waiting for more then 1 year already...
PS Will probably try to do some refactoring/splitting in nearest weeks within other PRs

**Pull request merged**